backend/inspirehep/records/marshmallow/literature/es.py

```diff
@@ -40,7 +42,16 @@ class LiteratureElasticSearchSchema(ElasticSearchBaseSchema, LiteratureRawSchema
     _bibtex_display = fields.Method("get_bibtex_display", dump_only=True)
     abstracts = fields.Nested(AbstractSource, dump_only=True, many=True)
     author_count = fields.Method("get_author_count")
-    authors = fields.Nested(AuthorsInfoSchemaForES, dump_only=True, many=True)
+    authors = NestedWithoutEmptyObjects(
+        AuthorsInfoSchemaForES,
+        default=[],
+        dump_only=True,
+        many=True,
+        attribute="authors",
+    )
+    supervisors = NestedWithoutEmptyObjects(
+        SupervisorSchema, default=[], dump_only=True, many=True, attribute="authors"
+    )
```

Comment:
How is this related to the rest? I thought we were already indexing the supervisors separately? or were we just removing them and not indexing them at all?

Reply:
we were just removing supervisors and I needed them indexed to look for supervisors affiliations